src/GafferSceneUI/SelectionTool.cpp

```diff
@@ -407,7 +407,12 @@ IECore::RunTimeTypedPtr SelectionTool::dragBegin( GafferUI::Gadget *gadget, cons
 	SceneGadget *sg = sceneGadget();
 	ScenePlug::ScenePath objectUnderMouse;
 
-	if( !sg->objectAt( event.line, objectUnderMouse ) )
+	// We treat drags as region selects ( instead of dragging the value of current selection ) if
+	// shift or control is held or the drag starts over the background.
+	if(
+		( event.modifiers & DragDropEvent::Shift ) || ( event.modifiers & DragDropEvent::Control ) ||
+		!sg->objectAt( event.line, objectUnderMouse )
+	)
```

Comment:
Sorry, I don't think I tested this thoroughly enough previously. We're still handling the click, so the object under the mouse gets selected/deselected before the drag starts, which kindof breaks the experience of the drag being a single gesture. The only way I can see of fixing that is to move the select/deselect to button release?